packages/common/src/fetchUtil.ts

```diff
@@ -67,46 +67,51 @@ function createDefaultMiddleware(): FetchMiddleware[] {
   return [setOriginMiddleware];
 }
 
-export function createFetchFn(fetchLib: FetchFn, ...middleware: FetchMiddleware[]): FetchFn;
-export function createFetchFn(...middleware: FetchMiddleware[]): FetchFn;
-export function createFetchFn(...args: any[]): FetchFn {
+// Argument helper function for {createFetchFn}
+function argsForCreateFetchFn(args: any[]): { middlewares: FetchMiddleware[]; fetchLib: FetchFn } {
   let fetchLib: FetchFn = fetch;
-  let middlewareOpt: FetchMiddleware[] = [];
-  if (args.length > 0) {
-    if (typeof args[0] === 'function') {
-      fetchLib = args.shift();
-    }
+  let middlewares: FetchMiddleware[] = [];
+  if (args.length > 0 && typeof args[0] === 'function') {
+    fetchLib = args.shift();
   }
   if (args.length > 0) {
-    middlewareOpt = args;
+    middlewares = args;
   }
-  const middlewares = [...createDefaultMiddleware(), ...middlewareOpt];
+  return { middlewares, fetchLib };
+}
+
+export function createFetchFn(fetchLib: FetchFn, ...middleware: FetchMiddleware[]): FetchFn;
+export function createFetchFn(...middleware: FetchMiddleware[]): FetchFn;
+export function createFetchFn(...args: any[]): FetchFn {
+  const { middlewares: middlewareArgs, fetchLib } = argsForCreateFetchFn(args);
+  const middlewares = [...createDefaultMiddleware(), ...middlewareArgs];
+
   const fetchFn = async (url: string, init?: RequestInit | undefined): Promise<Response> => {
     let fetchParams = { url, init: init ?? {} };
     for (const middleware of middlewares) {
-      if (middleware.pre) {
-        const result = await Promise.resolve(
-          middleware.pre({
-            fetch: fetchLib,
-            ...fetchParams,
-          })
-        );
-        fetchParams = result ?? fetchParams;
-      }
+      if (typeof middleware.pre !== 'function') continue;
+      const result = await Promise.resolve(
+        middleware.pre({
+          fetch: fetchLib,
+          ...fetchParams,
+        })
+      );
+      fetchParams = result ?? fetchParams;
```

Comment:
Nit: my code style preference would be something like
```js
if (typeof middleware.pre === 'function') {
   const result = await Promise.resolve(...);
}
```
Makes things a bit more readable by avoiding the coding equivalent of a "double negative", and avoid reasoning about early returns/continues.